Lib/profiling/sampling/live_collector/collector.py

```diff
@@ -0,0 +1,987 @@
+        msg1 = "WARNING: Terminal too small!"
+        msg2 = f"Required: {MIN_TERMINAL_WIDTH}x{MIN_TERMINAL_HEIGHT}"
+        msg3 = f"Current:  {width}x{height}"
+        msg4 = "Please resize your terminal for best experience"
+        msg5 = "Press any key to continue..."
+
+        # Center the messages
+        if height >= 5:
+            self.display.add_str(
+                height // 2 - 2,
+                max(0, (width - len(msg1)) // 2),
+                msg1[: width - 1],
+                A_BOLD,
+            )
+            self.display.add_str(
+                height // 2 - 1,
+                max(0, (width - len(msg2)) // 2),
+                msg2[: width - 1],
+            )
+            self.display.add_str(
+                height // 2,
+                max(0, (width - len(msg3)) // 2),
+                msg3[: width - 1],
+            )
+            self.display.add_str(
+                height // 2 + 1,
+                max(0, (width - len(msg4)) // 2),
+                msg4[: width - 1],
+            )
+            self.display.add_str(
+                height // 2 + 3,
+                max(0, (width - len(msg5)) // 2),
+                msg5[: width - 1],
+                A_DIM,
+            )
+        elif height >= 1:
+            self.display.add_str(0, 0, msg1[: width - 1], A_BOLD)
+
+        self.display.refresh()
+        # Wait for user acknowledgment (2 seconds timeout)
+        self.display.set_nodelay(False)
+        # Note: timeout is curses-specific, skipping for now
+        self.display.get_input()
+        self.display.set_nodelay(True)
+
+    def _handle_input(self):
+        """Handle keyboard input (non-blocking)."""
+        from . import constants
+
+        self.display.set_nodelay(True)
+        ch = self.display.get_input()
+
+        # If showing help, any key closes it
+        if self.show_help and ch != -1:
+            self.show_help = False
+            return
+
+        # Handle filter input mode
+        if self.filter_input_mode:
+            if ch == 27:  # ESC key
+                self.filter_input_mode = False
+                self.filter_input_buffer = ""
+            elif ch == 10 or ch == 13:  # Enter key
+                self.filter_pattern = (
+                    self.filter_input_buffer
+                    if self.filter_input_buffer
+                    else None
+                )
+                self.filter_input_mode = False
+                self.filter_input_buffer = ""
+            elif ch == 127 or ch == 263:  # Backspace
+                if self.filter_input_buffer:
+                    self.filter_input_buffer = self.filter_input_buffer[:-1]
+            elif ch >= 32 and ch < 127:  # Printable characters
+                self.filter_input_buffer += chr(ch)
+            return
+
+        # If finished, only allow 'q' to quit
+        if self.finished:
+            if ch == ord("q") or ch == ord("Q"):
+                self.running = False
+            return
```

Comment:
I think there's a bit of an awkward edge case here. I had help open when profiling ended (because I got enamored by all the options 😍 ) and I felt like I got stuck here. I kind of expected to be able to toggle help so I could view the snapshot again at the end before closing out.

I think we can account for this by supporting the toggle of h/H/? here, and adding an `_update_display()` in the loop after `finished`	 and then freezing the elapsed time?